src/contracts/pods/EigenPod.sol

```diff
@@ -635,8 +645,19 @@ contract EigenPod is Initializable, ReentrancyGuardUpgradeable, EigenPodPausingC
             delete currentCheckpointTimestamp;
             delete _currentCheckpoint;
 
+            // Calculate the slashing proportion
+            uint256 proportionPodBalanceDecrease = 0;
+            if (checkpoint.balanceDeltasGwei < 0) {
+                uint256 totalBefore = withdrawableRestakedExecutionLayerGwei + checkpoint.beaconChainBalanceBefore;
+                proportionPodBalanceDecrease = uint256(uint128(-checkpoint.balanceDeltasGwei)) * 1e18 / totalBefore;
+            }
```

Comment:
This seems to only care about the beacon chain portion of the delta, which is incorrect.

For example, if I withdraw a validator, I'll end up with a negative delta here - but I think you're trying to capture actual asset decrease, not just beacon chain balance decrease.

You need to capture the entire share delta

Reply:
you're right! patched